controllers/storagecluster/storageconsumer.go

```diff
@@ -57,39 +60,47 @@ func (s *storageConsumer) ensureCreated(r *StorageClusterReconciler, storageClus
 		return ctrl.Result{}, fmt.Errorf("failed to generate volumegroupsnapshotclasses list for distribution: %v", err)
 	}
 
-	storageConsumer := &ocsv1a1.StorageConsumer{}
-	storageConsumer.Name = defaults.LocalStorageConsumerName
-	storageConsumer.Namespace = storageCluster.Namespace
-	if _, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, storageConsumer, func() error {
-		if err := controllerutil.SetControllerReference(storageCluster, storageConsumer, r.Scheme); err != nil {
+	consumerName, err := getInternalStorageConsumerName(r, storageCluster.Name, storageCluster.Namespace)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get internal storage consumer name: %v", err)
+	}
+
+	storageConsumerObj := &ocsv1a1.StorageConsumer{}
+	storageConsumerObj.Name = consumerName
+	storageConsumerObj.Namespace = storageCluster.Namespace
+
+	if _, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, storageConsumerObj, func() error {
+		if err := controllerutil.SetControllerReference(storageCluster, storageConsumerObj, r.Scheme); err != nil {
 			return err
 		}
-		spec := &storageConsumer.Spec
-		spec.ResourceNameMappingConfigMap.Name = localStorageConsumerConfigMapName
+		spec := &storageConsumerObj.Spec
+		if spec.ResourceNameMappingConfigMap.Name == "" {
```

Comment:
is this check for upgrade controller could set this? if yes, both of the controllers can run in parallel, isn't it?